server/backup_binlog.go

```diff
@@ -0,0 +1,346 @@
+package server
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/cybozu-go/log"
+	"github.com/cybozu-go/moco"
+	mocoagent "github.com/cybozu-go/moco-agent"
+	"github.com/cybozu-go/well"
+	"github.com/jmoiron/sqlx"
+)
+
+// BackupBinaryLogsParams is the paramters for backup binary logs
+type BackupBinaryLogsParams struct {
+	FilePrefix   string
+	BucketHost   string
+	BucketPort   int
+	BucketName   string
+	BucketRegion string
+
+	AccessKeyID     string
+	SecretAccessKey string
+}
+
+// FlushAndBackupBinaryLogs executes "FLUSH BINARY LOGS;"
+// and upload it to the object storage, then delete it
+func (a *Agent) FlushAndBackupBinaryLogs(w http.ResponseWriter, r *http.Request) {
+	var err error
+	if r.Method != http.MethodGet {
```

Comment:
Why do you use GET method? POST is better because changes will be made. (by zoetrope)